asmcomp/arm64/emit.mlp

```diff
@@ -437,7 +438,11 @@ module BR = Branch_relaxation.Make (struct
     | Lop (Iextcall { alloc = true; }) -> 3
     | Lop (Istackoffset _) -> 2
     | Lop (Iload (size, addr)) | Lop (Istore (size, addr, _)) ->
-      let based = match addr with Iindexed _ -> 0 | Ibased _ -> 1 in
+      let based =
+        match addr with
+        | Iindexed _ -> 0
+        | Ibased _ -> if !Clflags.dlcode then 2 else 1
+      in
```

Comment:
Change not required if we ensure that `Ibased` mode is not used in `dlcode` mode.